src/libraries/AccountAccessParser.sol

```diff
@@ -335,19 +340,22 @@ library AccountAccessParser {
     }
 
     /// @notice Checks if the state diff represents setting an approve hash in a Gnosis Safe
-    function isGnosisSafeApproveHash(StateDiff memory _diff) internal pure returns (bool) {
-        // In Gnosis Safe, approvedHashes is a mapping at slot 8
-        // The slot for a specific hash approval is calculated with the keccak256 hash of the address
-        // and slot. We can't know the specific slot value here, but we can detect a change from 0 to 1
-        // which is a characteristic of an approve hash operation. We assume that if such a storage
-        // change is seen in a very large slot, it is an approve hash operation. Setting modules
-        // and guards in storage would not result in the slot having a value of 1.
-        // TODO Consider making this more robust to compute slots, if it adds value. In the safe
-        // storage layout, the only other mapping that sets slots to 1 is the `signedMessages`
-        // mapping but we don't use that. (The owners linked list can also have a value of 1 for the
-        // sentinel owner but that is only written when the safe is originally setup)
-        return _diff.oldValue == bytes32(0) && _diff.newValue == bytes32(uint256(1))
-            && _diff.slot > bytes32(uint256(0x1000000000));
+    function isGnosisSafeApproveHash(StateDiff memory _diff, address multisig, bytes32 _txHash)
+        internal
+        view
+        returns (bool)
+    {
+        bytes32[] memory hashSlots = calculateApproveHashSlots(IGnosisSafe(multisig).getOwners(), _txHash);
```

Comment:
Do we also need to add Multicall3 as an "owner" since we use use state overrides to make it an owner sometimes?

Reply:
We don't add the state variables for the local simulation, we only add those for the tenderly links. So no.